v2/internal/io/converter.go

```diff
@@ -106,6 +120,30 @@ func (drsb *discardingReadSeekerPlusByte) Seek(offset int64, whence int) (int64,
 	}
 }
 
+func (ras *readerAtSeeker) Read(p []byte) (n int, err error) {
+	ras.mu.Lock()
+	defer ras.mu.Unlock()
+	n, err = ras.ra.ReadAt(p, ras.position)
+	ras.position += int64(n)
+	return n, err
+}
+
+func (ras *readerAtSeeker) Seek(offset int64, whence int) (int64, error) {
+	ras.mu.Lock()
+	defer ras.mu.Unlock()
+	switch whence {
+	case io.SeekStart:
+		ras.position = offset
+	case io.SeekCurrent:
+		ras.position += offset
+	case io.SeekEnd:
+		panic("unsupported whence: io.SeekEnd")
+	default:
+		panic("unsupported whence")
```

Comment:
we can return errors  - can we do that instead of errors?

Reply:
We can, how strongly do you feel about it? What I was going for here was that if you encounter this then you're in programmer-error territory, you really shouldn't and you should know about it ASAP. That's normally the heuristic for choosing between disruptive and recoverable error types.

Reply:
if you end up with an edge case that isn't tested in the future that triggers a `SeekEnd`, you could end up with unexpected panics. Given how we end up with security problems when that happens, it seems safer to yield an error

Reply:
ok 7d8558c turns them into error returns and gives the same treatment to `discardingReadSeekerPlusByte` which is doing the same thing for unsupported operations